docs/topics/db/optimization.txt

```diff
@@ -411,4 +416,105 @@ objects to reduce the number of SQL queries. For example::
     my_band.members.add(me)
     my_band.members.add(my_friend)
 
-...where ``Bands`` and ``Artists`` have a many-to-many relationship.
+...where ``Band`` and ``Artist`` have a many-to-many relationship.
+
+If you have a list of objects, use the ``*``-operator when calling
+:meth:`~django.db.models.fields.related.RelatedManager.add`::
+
+    my_band.members.add(*new_members)
+
+Many source objects, many target objects
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When inserting pairs of objects into :class:`ManyToManyFields
+<django.db.models.ManyToManyField>`, use 
+:meth:`~django.db.models.query.QuerySet.bulk_create()` with multiple
+:attr:`through <django.db.models.ManyToManyField.through>` model instances 
+to reduce the number of SQL queries. For example::
```

Comment:
I would say:
```
When inserting different pairs of objects into
:class:`~django.db.models.ManyToManyField` or when the custom
:attr:`~django.db.models.ManyToManyField.through` table is defined, use
:meth:`~django.db.models.query.QuerySet.bulk_create()` method to reduce the
number of SQL queries. For example::

    Membership.objects.bulk_create([
        Membership(group=my_band, person=me, date_joined=date(1962, 8, 16)),
        Membership(group=my_band, person=my_friend, date_joined=date(1960, 8, 1)),
        Membership(group=they_band, person=my_sister, date_joined=date(1971, 8, 1)),
        Membership(group=they_band, person=my_brother, date_joined=date(1972, 1, 1)),
    ])
```

![obraz](https://user-images.githubusercontent.com/2865885/67678725-a6738300-f987-11e9-9395-5ddc82e0f515.png)

Reply:
I think for both `Insert` and `Delete` sections a single example is enough. 

I'd be inclined to use `Pizza.toppings.through` in both cases (with `ignore_conflicts=True` for Insert). (Assuming folks with an explicit through model will know to use that...)